### PR TITLE
Next generation Sigma clipping implemented in OpenCL

### PR DESCRIPTION
**Pull request by kif**
This is untested for now. to be validated

### CODE REVIEW

File: pyFAI/resources/openCL/ocl_azim_CSR.cl [hunk 1]
```diff
@@ -301,6 +301,65 @@ static inline float8 CSRxVec4(const   global  float4   *data,
...
+	barrier(CLK_LOCAL_MEM_FENCE);
+    for (j=bin_bounds.x; j<bin_bounds.y; j+=WORKGROUP_SIZE)
+    {
+        k = j+thread_id_loc;
+        if (k < bin_bounds.y)
+        {
+        	idx = indices[k];
+            float4 quatret = data[idx];
+            // Check validity (on cnt, i.e. s3) and normalisation (in s2) value to avoid zero division 
+            if (isfinite(quatret.s3) & (quatret.s2>0.0f))
```
Comment (pierrepaleo) [inline pyFAI/resources/openCL/ocl_azim_CSR.cl]: would be better with `&&`

File: pyFAI/azimuthalIntegrator.py [hunk 2]
```diff
@@ -3254,6 +3292,58 @@ def sigma_clip(self, data, npt_rad=1024, npt_azim=512,
...
+        :param unit: unit to be used for integration
+        :param method: pathway for integration and sort
+        :param thres: cut-off for n*sigma: discard any values with (I-<I>)/sigma > thres.
+                The threshold can be a 2-tuple with sigma_low and sigma_high.
+        :param max_iter: maximum number of iterations        :param mask: masked out pixels array
+        :param float normalization_factor: Value of a normalization monitor
+        :param metadata: any other metadata,
+        :type metadata: JSON serializable dict
+        :return: Integrate1D like result like
+
```
Comment (pierrepaleo) [inline pyFAI/azimuthalIntegrator.py]: Some typos/incomplete documentation below

**Comment by kif** [comment]
I will leave it like this:
Sigma clipping works as expected. Peak-finder is ongoing work but will be continued in another PR

**Comment by kif** [comment]
close  #1285
Waiting for proper exposition in azimuthal integrator

**Comment by kif** [comment]
Remains to be done:
- [ ] the selection of the method is not working
- [ ] setup_CSR() got an unexpected keyword argument 'radial_range'
- [ ] no test basically check that it works

**Comment by kif** [comment]
```
ERROR: test_1d_splitbbox (pyFAI.test.test_csr.TestCSR)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/users/kieffer/workspace-400/pyFAI/build/lib.linux-x86_64-3.7/pyFAI/test/test_csr.py", line 121, in test_1d_splitbbox
    bin_centers=engine.bin_centers)
TypeError: __init__() got an unexpected keyword argument 'data'
```

**Comment by kif** [comment]
All tests are passing ... merging

**Pull request merged**